staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/strategy.go

```diff
@@ -334,6 +334,9 @@ func dropDisabledFields(newCRD *apiextensions.CustomResourceDefinition, oldCRD *
 	if !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceFieldSelectors) && (oldCRD == nil || (oldCRD != nil && !specHasSelectableFields(&oldCRD.Spec))) {
 		dropSelectableFields(&newCRD.Spec)
 	}
+	if !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CRDObservedGenerationTracking) {
```

Comment:
also check if the old object already had these fields set to non-zero values before dropping... this is what will allow a 1.35 server to tolerate/preserve data allowed into etcd by a 1.36 server

```suggestion
	if !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CRDObservedGenerationTracking) && !observedGenerationTrackingInUse(oldCRD) {
```